ir/src/constraints/graph.rs

```diff
@@ -398,6 +423,14 @@ impl AlgebraicGraph {
                 Ok(ExprDetails::new(node_index, trace_segment, domain))
             }
             IdentifierType::PublicInput(_) => {
+                if !matches!(
+                    domain,
+                    ConstraintDomain::FirstRow | ConstraintDomain::LastRow
```

Comment:
Should we add `is_boundary()` to `ConstraintDomain`?

Reply:
Done